### PR TITLE
Display loader when site title is updating

### LINKED ISSUE #15717: When changing Site Title. there is no indication that something is updating
### Expected behavior
When I change my site title, if it doesn't happen immediately, there should be indication that it is updating. 

### Actual behavior
There is no indication. 
While checking Android, the change was instant so no indication of title changing seems necessary. 
On iOS, it took less than 2 seconds but wasn't instant. 
I have seen it take about 5 seconds before on a slower connection with no indication that it's changing.

### Steps to reproduce the behavior
on the My Site page:
1. Tap the title of your site under the icon
2. A Site Title modal will open where you can edit.
3. Make some change and Save

### PR DESCRIPTION
**Pull request by yaelirub**
Fixes #15717 

To test:
on the My Site page:

1. Tap the title of your site under the icon
2. A Site Title modal will open where you can edit.
3. Make some change and Save
4. See loader 
5. See loader removed when stopped loading 
6. See title updated 

https://user-images.githubusercontent.com/1335657/107131474-6b027c80-688b-11eb-98bb-d21833a9bc69.mov

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/Blog Details/Detail Header/BlogDetailHeaderView.swift [hunk 1]
```diff
@@ -92,6 +92,10 @@ class BlogDetailHeaderView: UIView, BlogDetailHeader {
         siteIconView.spotlightIsShown = QuickStartTourGuide.shared.isCurrentElement(.siteIcon)
     }
 
+    @objc func updateLoadingTitle(isLoading: Bool) {
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Blog/Blog Details/Detail Header/BlogDetailHeaderView.swift]: How about `setTitleLoading(_ isLoading: Bool)`? (Would prevent the duplication of 'loading' when calling it, and I think `setTitleLoading` may be a little clearer than the existing about what it's doing?)

File: WordPress/Classes/Utility/Spotlight/Spotlightable.swift [hunk 2]
```diff
@@ -23,6 +23,28 @@ class SpotlightableButton: UIButton, Spotlightable {
         }
     }
 
+    func startLoading() {
+        setTitle("", for: .normal)
+        activityIndicator.startAnimating()
+    }
+
+    func stopLoading() {
+        activityIndicator.stopAnimating()
```
Comment (frosty) [inline WordPress/Classes/Utility/Spotlight/Spotlightable.swift:34]: Should we restore the title when we stop loading?

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+Header.swift [hunk 3]
```diff
@@ -77,6 +78,7 @@ extension BlogDetailsViewController {
                                 feedbackType: .success)
             ActionDispatcher.global.dispatch(NoticeAction.post(notice))
 
+            self?.headerView.updateLoadingTitle(isLoading: false)
             self?.headerView.refreshSiteTitle()
         }, failure: { [weak self] error in
             self?.blog.settings?.name = existingBlogTitle
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+Header.swift:84]: The header view is currently not updated if there's a failure with the request to update the settings, which results it showing the loading indicator indefinitely. We should also stop loading here and restore the original title.

**Review by frosty — COMMENTED** [review]
Mostly looking good! 🙂 Just one or two small comments, mainly handling the error case.

There's also a kind-of-edge-case which could happen if the user has a poor connection – if you navigate to another site and then back while the request is ongoing, when you come back there's nothing to indicate that it's happening. But I think it's fine to leave that case.

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+Header.swift (hunk 3 above)
Comment (yaelirub) [reply WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+Header.swift:84]: Absolutely! Thank you!

File: WordPress/Classes/ViewRelated/Blog/Blog Details/Detail Header/BlogDetailHeaderView.swift (hunk 1 above)
Comment (yaelirub) [reply WordPress/Classes/ViewRelated/Blog/Blog Details/Detail Header/BlogDetailHeaderView.swift]: Done in 1e49f70d7941a7b87ab0c07795ec71643224fcac

File: WordPress/Classes/Utility/Spotlight/Spotlightable.swift (hunk 2 above)
Comment (yaelirub) [reply WordPress/Classes/Utility/Spotlight/Spotlightable.swift:34]: Done in 1e49f70d7941a7b87ab0c07795ec71643224fcac

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+Header.swift (hunk 3 above)
Comment (yaelirub) [reply WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+Header.swift:84]: Done in 1e49f70d7941a7b87ab0c07795ec71643224fcac

**Comment by yaelirub** [comment]
@frosty , thank you for reviewing! Ready for another look

File: WordPress/Classes/Utility/Spotlight/Spotlightable.swift [hunk 4]
```diff
@@ -7,6 +7,7 @@ protocol Spotlightable: UIView {
 class SpotlightableButton: UIButton, Spotlightable {
 
     var spotlight: QuickStartSpotlightView?
+    var originalTtile: String?
```
Comment (frosty) [inline WordPress/Classes/Utility/Spotlight/Spotlightable.swift]: Typo:

```suggestion
    var originalTitle: String?
```

File: WordPress/Classes/Utility/Spotlight/Spotlightable.swift [hunk 5]
```diff
@@ -23,6 +24,30 @@ class SpotlightableButton: UIButton, Spotlightable {
         }
     }
 
+    func startLoading() {
+        originalTtile = titleLabel?.text
```
Comment (frosty) [inline WordPress/Classes/Utility/Spotlight/Spotlightable.swift]:
```suggestion
        originalTitle = titleLabel?.text
```

File: WordPress/Classes/Utility/Spotlight/Spotlightable.swift [hunk 6]
```diff
@@ -23,6 +24,30 @@ class SpotlightableButton: UIButton, Spotlightable {
...
+        setTitle("", for: .normal)
+        activityIndicator.startAnimating()
+    }
+
+    func stopLoading() {
+        activityIndicator.stopAnimating()
+        setTitle(originalTtile, for: .normal)
```
Comment (frosty) [inline WordPress/Classes/Utility/Spotlight/Spotlightable.swift]:
```suggestion
        setTitle(originalTitle, for: .normal)
```

**Review by frosty — COMMENTED** [review]
Thanks for making the changes! Just one minor typo – I've added some suggestions so you should be able to just amend it via Github.

Other than that, looking great 👍

**Comment by yaelirub** [comment]
Good catch, @frosty! I committed your suggestions

**Review by frosty — APPROVED** [review]
Looks great! ✅

**Pull request merged**